pkg/cli/add_vcluster_helm.go

```diff
@@ -33,22 +39,105 @@ func AddVClusterHelm(
 	vClusterName string,
 	log log.Logger,
 ) error {
-	// check if vCluster exists
-	vCluster, err := find.GetVCluster(ctx, globalFlags.Context, vClusterName, globalFlags.Namespace, log)
-	if err != nil {
-		return err
+	var vClusters []find.VCluster
+	if options.All {
+		log.Debugf("add vcluster called with --all flag")
+		kubeClientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{
+				CurrentContext: globalFlags.Context,
+			})
+		hostClusterRestConfig, err := kubeClientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		hostKubeClient, err := kubernetes.NewForConfig(hostClusterRestConfig)
+		if err != nil {
+			return err
+		}
+		namespaces, err := hostKubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		log.Debugf("looking for vclusters in %d namespaces", len(namespaces.Items))
+		for _, ns := range namespaces.Items {
+			log.Infof("looking for a vcluster in %s namespace", ns.GetName())
+			vClustersInNamespace, err := find.ListVClusters(ctx, globalFlags.Context, "", ns.GetName(), log)
+			if err != nil {
+				return err
+			}
+			if len(vClustersInNamespace) == 0 {
+				log.Infof("no vClusters found in context %s and namespace %s", globalFlags.Context, ns.GetName())
+				continue
+			}
+			vClusters = append(vClusters, vClustersInNamespace...)
+		}
+	} else {
+		// check if vCluster exists
+		vCluster, err := find.GetVCluster(ctx, globalFlags.Context, vClusterName, globalFlags.Namespace, log)
+		if err != nil {
+			return err
+		}
+		vClusters = append(vClusters, *vCluster)
 	}
 
-	// create kube client
-	restConfig, err := vCluster.ClientFactory.ClientConfig()
+	if len(vClusters) == 0 {
+		return nil
+	}
+
+	restConfig, err := vClusters[0].ClientFactory.ClientConfig()
 	if err != nil {
 		return err
 	}
+
+	// create kube client
 	kubeClient, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
 		return err
 	}
+	addErr := &VClusterAddError{}
+	log.Debugf("trying to add %d vClusters to platform", len(vClusters))
+	for _, vCluster := range vClusters {
+		vCluster := vCluster
+		log.Infof("adding %s vCluster to platform", vCluster.Name)
+		addErr.addErr(vCluster.Name, addVClusterHelm(ctx, options, globalFlags, vCluster.Name, &vCluster, kubeClient, log))
+	}
+
+	return addErr.CombinedError()
+}
+
+type VClusterAddError struct {
+	errs []error
+}
+
+func (vce *VClusterAddError) CombinedError() error {
+	if len(vce.errs) == 0 {
+		return nil
+	} else if len(vce.errs) == 1 {
+		return vce.errs[0]
+	}
+	errMsg := strings.Builder{}
+	for _, err := range vce.errs {
+		_, _ = errMsg.WriteString(err.Error() + "|")
+	}
+	return errors.New(errMsg.String())
+}
 
+func (vce *VClusterAddError) addErr(vClusterName string, err error) {
+	if err == nil {
+		return
+	}
+	vce.errs = append(vce.errs, fmt.Errorf("cannot add vcluster %s: %w", vClusterName, err))
+}
```

Comment:
For aggregating errors, use a pattern like [this](https://github.com/loft-sh/vcluster/blob/cf9a1a48a2b38594df2c4139bbb1d62f23f9bf69/pkg/controllers/resources/nodes/nodeservice/node_service.go#L83-L114) instead. TL;DR append them to a slice and then use `utilerrors.NewAggregate`.